lib/doc.js

```diff
@@ -67,11 +71,12 @@ var markedOptions = {
  * @constructor
  */
 
-function Doc(file, contents, isJS, docs) {
+function Doc(file, contents, isJS, isTS, docs) {
```

Comment:
boolean arguments are difficult to understand when reading the calling code.

```ts
new Doc(file, contents, false, true, docs)
```

Also the constructor has already too many arguments, adding more of them does not scale well.

I am proposing the following possible improvements:

 1. `function Doc(file, contents, flags, docs)` where `isJS = flags.isJS` and `isTS = flags.isTS`.
 2. Alternatively, replace all arguments with a single options object `function Doc(options)`.
 3. Or perhaps change `isJS` to `fileExtension` and then `isJS = fileExtension === '.js'`, etc.